src/basicfuns.jl

```diff
@@ -309,9 +309,22 @@ Return `log(exp(x) + exp(y))`, avoiding intermediate overflow/undeflow, and hand
 non-finite values.
 """
 function logaddexp(x::Real, y::Real)
-    # ensure Δ = 0 if x = y = ± Inf
-    Δ = x == y ? zero(x - y) : abs(x - y)
-    max(x, y) + log1pexp(-Δ)
+    if isnan(x) return x end
+    if isnan(y) return y end
```

Comment:
This causes type instabilities and can't be used unfortunately. It's bad that this is not covered by our tests, it seems.